internal/consensus/state.go

```diff
@@ -2090,9 +2090,15 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal, recvTime time.Time
 	// TODO: We can check if Proposal is for a different block as this is a sign of misbehavior!
 	if cs.ProposalBlockParts == nil {
 		cs.ProposalBlockParts = types.NewPartSetFromHeader(proposal.BlockID.PartSetHeader)
+
+		// If we are the proposer, lock the PartSet until we load all
+		// the block parts, that should come just after this Proposal.
+		if bytes.Equal(proposerPubKey.Bytes(), cs.privValidatorPubKey.Bytes()) {
```

Comment:
```suggestion
		if cs.isProposer(cs.privValidatorPubKey.Address()) {
```

Reply:
I have accepted it then reverted it.

We have to test if the `Proposal` has been signed by us.